src/server.ts

```diff
@@ -15,12 +15,12 @@ export const executeAsIfOnServer = async <T>(f: () => T) => {
   // @ts-ignore
   delete global.document;
 
-  const result = await f();
-
-  global.window = tmpWindow;
-  global.document = tmpDocument;
-
-  return result;
+  try {
+    return await f();
+  } finally {
+    global.window = tmpWindow;
+    global.document = tmpDocument;
```

Comment:
Not really related to this fix, but I noticed this is an issue when experimenting with custom `_error` page. When an `Error` is thrown from some data fetching method we didn't restore `client` environment which lead to crash at render time, because `document`  was not defined.

Reply:
I had the impression something like this was happening yesterday when I had several broken tests. Clever idea 👍